hera_sim/reflections.py

```diff
@@ -0,0 +1,82 @@
+"""A module for generating reflections"""
+
+import numpy as np
+import aipy
+
+def auto_reflection(vis, freqs, amp, dly, phs, conj=False):
+    """
+    Insert an auto reflection into vis data. An auto reflection
+    is one that couples an antenna's voltage stream with itself.
+    Examples include cable reflections and dish-to-feed reflections.
+
+    Args:
+        vis : 2D complex ndarray, shape=(Ntimes, Nfreqs)
+        freqs : 1D ndarray, holds frequencies in GHz
+        amp : float, reflection amplitude
+        dly : float, reflection delay [nanosec]
+        phs : float, reflection phase [radian]
+        conj : bool, if True, conjugate reflection coefficient
+
+    Returns:
+        out_vis : contains input vis with the auto reflection
+    """
+    # form reflection coefficient
+    eps = amp * np.exp(2j * np.pi * freqs * dly + 1j * phs)
+    if conj:
+        eps = eps.conj()
+
+    # multiply into data
+    out_vis = vis * (1 + eps)
+
+    return out_vis
+
+
+def cross_reflection(vis, freqs, autocorr, amp, dly, phs, conj=False):
+    """
```

Comment:
From what is in this documentaiton, its confusing as to why this situation is distinct from the ordinary reflection above -- if the source of the cross reflection is coming from the sky --. I suspect that you are referring to reflections of non-sky-signals such as antenna thermal noise? If so, I think you should say this explicitly in the documentation.

Reply:
I added some math to make it more clear hopefully. I do mean reflections of "signal" (any kind of signal, noise or sky or anything else).

Reply:
Thanks for clarifying this!